code/test/VsTemplates.Test/VsTemplatesContentValidationTests.cs

```diff
@@ -0,0 +1,136 @@
+        private const string ProjectTemplatesRoot = "../../../../../src/ProjectTemplates";
+        private const string VsTemplateFileExtension = ".vstemplate";
+        private static readonly List<string> WizardProjectTemplatesRoot = new List<string>
+        {
+            "../../../../../src/ProjectTemplates/UWP",
+            "../../../../../src/ProjectTemplates/WPF",
+            "../../../../../src/ProjectTemplates/WinUI/Cpp/Cpp.WinUI.Desktop.Solution",
+            "../../../../../src/ProjectTemplates/WinUI/Cpp/Cpp.WinUI.UWP.Solution",
+            "../../../../../src/ProjectTemplates/WinUI/CS/CSharp.WinUI.Desktop.Solution",
+            "../../../../../src/ProjectTemplates/WinUI/CS/CSharp.WinUI.Uwp.Solution",
+        };
+
+        public static IEnumerable<object[]> GetWizardProjectTemplateFiles()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            foreach (var path in WizardProjectTemplatesRoot)
+            {
+                var filePaths = Directory.GetFiles(path, $"*{VsTemplateFileExtension}", SearchOption.AllDirectories);
+                vsTemplatesPaths.AddRange(filePaths.Select(p => new object[] { p }));
+            }
+
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsTemplatePaths()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            var vsItemTemplatesPaths = GetAllVsTemplatePaths(ItemTemplatesRoot).Select(p => new object[] { p });
+            var vsProjectTemplatesPaths = GetAllVsTemplatePaths(ProjectTemplatesRoot).Select(p => new object[] { p });
+            vsTemplatesPaths.AddRange(vsItemTemplatesPaths);
+            vsTemplatesPaths.AddRange(vsProjectTemplatesPaths);
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsTemplateFiles()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            var vsItemTemplatesPaths = GetAllVsItemTemplateFiles();
+            var vsProjectTemplatesPaths = GetAllVsProjectTemplateFiles();
+            vsTemplatesPaths.AddRange(vsItemTemplatesPaths);
+            vsTemplatesPaths.AddRange(vsProjectTemplatesPaths);
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsProjectTemplateFiles()
+        {
+            var filePaths = Directory.GetFiles(ProjectTemplatesRoot, $"*{VsTemplateFileExtension}", SearchOption.AllDirectories);
+            return filePaths.Select(p => new object[] { p });
+        }
+
+        private static IEnumerable<object[]> GetAllVsItemTemplateFiles()
+        {
+            var filePaths = Directory.GetFiles(ItemTemplatesRoot, $"*{VsTemplateFileExtension}", SearchOption.AllDirectories);
+            return filePaths.Select(p => new object[] { p });
+        }
+
+        private static IEnumerable<string> GetAllVsTemplatePaths(string directory)
+        {
+            foreach (var dir in Directory.GetDirectories(directory))
```

Comment:
Rather than loop through the directories here, why not have GetFiles recurse? 
Or is your code doing something else beyond explicitly recursing the directory structure? If it is it, the reason for this should be documented in the code.

Reply:
@sibille What was the reason for this manual recursion?

Reply:
The test needs the parent folder name of the vstemplate files to compare localized files with the original vstemplate. 
IMHO with the name change @vmadurga did the intention of what the method does clear now.
Are you seeing an easier way to do this with Directory.GetFiles()?